flik/linesearch/conditions.py

```diff
@@ -1 +1,47 @@
 """Conditions for the line search algorithm."""
+def soft_wolfe(gradf, valf, alpha, direction, const2=0.9):
+    """Curvature condition
+
+    Parameters
+    ----------
+    gradf: function
+        Gradient of objective nonlinear function
+    valf: np.ndarray(N,)
+        Function value
+    alpha: float
+        Step length
+    direction: np.ndarray(N,)
+        Descent direction
+    const2: float
+        Constant C2
+
+    Raises
+    ------
+    ValueError
+        If C2 is not between 0 and 1
+    TypeError
+        If gradient function if not a function
+
+    Returns
+    -------
+    Boolean
+        True if curvature condition is satisfied
```

Comment:
I'm not sure if `Boolean` is necessary. Could you also include `False otherwise`?

Reply:
Nevermind, the type would be necessary, but can it be changed to `bool` instead?